### PR TITLE
Move from jshint to eslint

### PR DESCRIPTION
**Pull request by Ladsgroup**
Also fix travis by pinpointing to phpunit 4.8

Bug: T165843

### CODE REVIEW

File: js/dataTypes/DataType.js [hunk 1]
```diff
@@ -17,63 +17,63 @@
...
 	/**
 	 * DataType identifier.
 	 * @property {string}
 	 * @private
 	 */
-	_id: null,
+		_id: null,
 
-	/**
+		/**
```
Comment (manicki) [inline js/dataTypes/DataType.js]: this look odd now

File: .travis.yml [hunk 2]
```diff
@@ -15,6 +18,8 @@ matrix:
 
 before_script:
   - composer install --prefer-source
+  - npm install eslint eslint-config-wikimedia
+  - wget https://phar.phpunit.de/phpunit-4.8.phar
```
Comment (manicki) [inline .travis.yml]: I guess it would be good to pin PHPUnit version in composer.json. I am adding a commit to this PR.

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+	"extends": "wikimedia",
+	"env": {
+		"browser": true,
+		"jquery": true
+	},
+	"globals": {
+		"dataTypes": false,
+		"define": false,
+		"QUnit": false
```
Comment (manicki) [inline .eslintrc.json]: I believe it is better to have test-specifc rules, environments, globals etc defined in test-only eslintrc file. I'll add a commit to the PR to show what I mean.

**Comment by manicki** [comment]
Ended up doing several changes. I hope you don't mind.

**Comment by Ladsgroup** [comment]
Not at all. Thanks!

**Pull request merged**